### PR TITLE
profiling: add tool path settings in project properity for profiling

### PR DESCRIPTION
**Pull request by laomaolaile**
Try to resolve issue #181

### CODE REVIEW

**Comment by jjohnstn** [comment]
Hello @laomaolaile To inevitably get this accepted you will need to sign the Eclipse Contributor Agreement for the e-mail you have attached to your github id.  Details can be found here: https://www.eclipse.org/legal/ECA.php  You will need to create an Eclipse account first if you haven't already.

**Review by jjohnstn — CHANGES_REQUESTED** [review]
Looks good.  A few issues to resolve which I have commented.

**Comment by jjohnstn** [comment]
Test results are expected with rpm and createrepo tests.  These are being removed.

**Comment by jjohnstn** [comment]
@laomaolaile If you can get the requested changes in before Tuesday of next week, your patch can get into 2023-09 M1.  Otherwise, it will have to wait until 2023-09 M2.

**Comment by laomaolaile** [comment]
@jjohnstn  Okay, I'll finish it this week

File: profiling/org.eclipse.linuxtools.tools.launch.core/src/org/eclipse/linuxtools/tools/launch/core/factory/LinuxtoolsProcessFactory.java [hunk 1]
```diff
@@ -41,7 +44,9 @@ public abstract class LinuxtoolsProcessFactory {
 
     private static final String PATH = "PATH"; //$NON-NLS-1$
     private static final String PATH_EQUAL = "PATH="; //$NON-NLS-1$
-    private static final String SEPARATOR = ":"; //$NON-NLS-1$
+	private static final String SEPARATOR_WIN32 = ";"; //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (jjohnstn) [reply profiling/org.eclipse.linuxtools.tools.launch.core/src/org/eclipse/linuxtools/tools/launch/core/factory/LinuxtoolsProcessFactory.java]: Looks good now.

File: profiling/org.eclipse.linuxtools.tools.launch.ui/src/org/eclipse/linuxtools/internal/tools/launch/ui/properties/messages.properties [hunk 2]
```diff
@@ -18,3 +18,9 @@ LINUXTOOLS_PATH_TOOLTIP=Use ':' as Path separator
 LINUXTOOLS_PATH_CUSTOM=Prepend string to PATH
 LINUXTOOLS_PATH_SYSTEM_ENV=Use the System Environment PATH
 LINUXTOOLS_PATH_CUSTOM_TOOLTIP=Append the selected string to the beginning of the current system Path
+LINUXTOOLS_PREFIX=
+LINUXTOOLS_PREFIX_COMBO=Prefix:
+LINUXTOOLS_PREFIX_TOOLTIP=Add extra prefix to command such as gcov addr2line nm c++filt strings gprof, eg, prefix riscv-nuclei-elf-, command, strings will be riscv-nuclei-elf-strings
```
[the remark this answers is not in the document]
Comment (jjohnstn) [reply profiling/org.eclipse.linuxtools.tools.launch.ui/src/org/eclipse/linuxtools/internal/tools/launch/ui/properties/messages.properties]: Change looks good.

**Review by jjohnstn — APPROVED** [review]

**Comment by jjohnstn** [comment]
Test failures have nothing to do with this patch.

**Comment by jjohnstn** [comment]
Thanks @laomaolaile

**Comment by fanghuaqi** [comment]
@laomaolaile You are using tab not spaces to indent code, and I found similar lines need to be adjusted. Maybe a new patch could be added to solve this

image
image
image

**Comment by jjohnstn** [comment]
@fanghuaqi Yes, he has used tabs which actually is the right way to do it.  The old code needs to be reformatted.  I noticed this in reviewing but did not want extra reformatting along with the patch as it obfuscates the actual changes.  Another formatting patch is required.  I was going to do it prior to 2023-09 M1, but feel free if you wish.

**Pull request merged**